### ISSUE TITLE
OK & CANCEL button placement

### ISSUE THREAD

**Issue by cooperzoe** [OpenRefine/OpenRefine#6353]
In the newest version of OpenRefine, I notice the OK button is on the left hand side and the CANCEL button on the right in this window. I expected the OK button to be on the right hand side and found it's unusual placement counterintuitive.

I propose we either switch their placements, or put OK in the righthand corner and keep cancel next to it. If we choose the latter option, perhaps we could make the OK button blue, so they don't look too similar to one another.

<img width="1177" alt="Screenshot 2024-02-08 at 12 31 38 PM" src="https://github.com/OpenRefine/OpenRefine/assets/52141007/bd5e1f3b-0684-4b57-9db4-13e56c6ab3ba">

**Comment by wetneb** [comment]
I would be in favor of agreeing one one convention and normalize all dialogs to follow that convention.

My vote goes to have both buttons on the right-hand side and the OK button to be marked as "primary" (meaning that its text will appear in bold).

**Comment by tfmorris** [comment]
I agree that the current placement is unusual. I think the application should follow the UI conventions of the underlying platform. They all have style guides and standard dialogs which implement those style guides (and perhaps they're all the same, but if not I think we should track the differences).

**Comment by areebqazi** [comment]
Hello is anyone doing this issue or can I work on it ?

**Comment by tfmorris** [comment]
@areebqazi Thanks for you interest, but we still need to come up with a firm design for this before it's ready for work.

**Comment by tfmorris** [comment]
Although Jakob Nielsen says to follow platform conventions, except for web interfaces, I'd argue that we actually have enough information available about the user environment that we're closer to the platform case than the "web" case. He also has some input on button styling.

https://www.nngroup.com/articles/ok-cancel-or-cancel-ok/
https://www.lukew.com/ff/entry.asp?571
https://ux.stackexchange.com/questions/1072/ok-cancel-on-left-right

**Comment by vr-varad** [comment]
so are the changes required in the buttons??

**Comment by wetneb** [comment]
@tfmorris do you mean we should detect the operating system and adapt the button placement based on that?
I'd rather go with a uniform style regardless of the OS, for simplicity.

**Comment by tfmorris** [comment]
> do you mean we should detect the operating system and adapt the button placement based on that? 

Yes, I think that would give the best user experience.

> I'd rather go with a uniform style regardless of the OS, for simplicity.

I understood that from your previous comment, but wasn't able to tell which style you were proposing. As I understand Zoe's proposal, she is proposing the Mac button ordering. Everyone agrees that all the buttons should go in the lower right corner, so, if using an order that the user expects is too complicated, we just need to choose between Windows (OK / Cancel) or Mac (Cancel / OK) ordering. Given that our Windows downloads are 2-3x the Mac downloads, I guess that means the default choice should be Windows.

**Comment by cooperzoe** [comment]
Hi everyone - apologies for the slow reply, I was out sick last week. Thank you for your input, @tfmorris @wetneb and everyone!

I've mocked up two options as @tfmorris has helpfully laid them out: one in Windows (OK / Cancel) ordering, and one in Mac (Cancel / OK) ordering. I hope the comparison is helpful to see:
<img width="542" alt="Screenshot 2024-02-26 at 6 33 15 PM" src="https://github.com/OpenRefine/OpenRefine/assets/52141007/e2ee4e0f-cda0-4e92-951f-48ce939b7227">
<img width="644" alt="Screenshot 2024-02-26 at 6 31 24 PM" src="https://github.com/OpenRefine/OpenRefine/assets/52141007/48ab8cb6-cef1-404e-b949-f6a4d167e42c">

**Comment by wetneb** [comment]
I'd propose to mark this as a good first issue, let's say with the Windows ordering ("OK" / "Cancel") for the sake of making a decision. The task would be to go through all dialogs and adjust them to this convention. It's a task that's a perhaps a bit harder than some other good first issues given that it will require a few more changes, but at least it should not require writing any test and is only requiring front-end changes.

**Comment by jquartel** [comment]
For what it's worth, I keep finding myself accidentally cancelling my custom facets, now, because I was used to 'OK' being on the right, so I'm in favour of moving 'OK' back to that region. I'm a Mac user and I didn't mind the OK/Cancel ordering we had before.

**Comment by zyadtaha** [comment]
Hi @wetneb,
I searched about this attribute `bind="okButton"` but it seems to be a custom attribute. 
If I am right, can you tell me what it refers to ?

**Comment by tfmorris** [comment]
> I was used to 'OK' being on the right, so I'm in favour of moving 'OK' back to that region.

I thought the current design was odd, but until @jquartel mentioned it, I didn't consciously realize it had been changed.

It looks like at least part of the problem is the styling is defined in multiple places, probably unintentionally. When I look in my browser, I see that `.dialog-footer` is being defined in `dialog.css`, which I'd expect, but also `recon-dialog.css` (from f856713fd81af6629178e38d93fe55016529204b) and `column-join.css` (27d5fb88fa9e0d2d941d3681e1f3363e703068a7) which are likely unintentional and which are overriding the base definition.

I think removing this override will fix the problem (or at least restore the previous appearance):
https://github.com/OpenRefine/OpenRefine/blob/f856713fd81af6629178e38d93fe55016529204b/main/webapp/modules/core/styles/reconciliation/recon-dialog.css#L105-L108

It looks like the problem was introduced just a few months ago, so the 3.8 beta is the only release it appears in.

**Comment by zyadtaha** [comment]
Awesome! so we to remove this line `justify-content: space-between; `.
but how to add the `style="font-weight: bold;"` to the Ok Button ? 
Should I put the style in every button tag in HTML or just add a class attribute to all ok buttons and make css to the class ?

**Comment by zyadtaha** [comment]
@wetneb Could you assign this to me ?

**Comment by wetneb** [comment]
I'm adding this commit to the 3.8 branch as @tfmorris mentioned that the problem was recent, hence worth fixing before the next stable release.

**Comment by tfmorris** [comment]
@wetneb These changes 

https://github.com/OpenRefine/OpenRefine/blob/f856713fd81af6629178e38d93fe55016529204b/main/webapp/modules/core/styles/reconciliation/recon-dialog.css#L105-L117

from f856713fd81af6629178e38d93fe55016529204b should probably be reviewed for removal as well. The CSS for the recon dialog shouldn't be messing with common styles.
